src/include/execution/exec/output.h

```diff
@@ -90,7 +91,7 @@ class EXPORT OutputBuffer {
   uint32_t num_tuples_;
   uint32_t tuple_size_;
   byte *tuples_;
-  OutputCallback callback_;
+  const OutputCallback &callback_;
```

Comment:
Could you document the conversation we had about const &'ing the outputcallback here?

Reply:
While const &'ing the OutputCallback is fine here, the way OutputCallback is setup might need to change. From a local build of master, it would appear that `std::function<>` copies the `OutputWriter` thus leading to an incorrect `OutputWriter::NumRows()` count (i.e always 0).

(comment here for reference)